ambry-frontend/src/main/java/com/github/ambry/frontend/DeleteBlobHandler.java

```diff
@@ -165,10 +167,13 @@ private Callback<Void> routerCallback() {
           }
         }
         LOGGER.debug("Deleted {}", getRequestPath(restRequest).getOperationOrBlobId(false));
-        restResponseChannel.setStatus(ResponseStatus.Accepted);
-        restResponseChannel.setHeader(RestUtils.Headers.DATE, new GregorianCalendar().getTime());
-        restResponseChannel.setHeader(RestUtils.Headers.CONTENT_LENGTH, 0);
-        securityService.processResponse(restRequest, restResponseChannel, null, securityProcessResponseCallback());
+        if (!(RequestPath.matchesOperation(restRequest.getUri(), ACCOUNTS_CONTAINERS_DATASETS)
```

Comment:
If it's dataset deletion, why do we need skip setheader and processResponse?
Can you add comment?

Reply:
Updated the comments. we should let DeleteDatasetHandler to handle it.